client/components/seo/meta-title-editor/index.jsx

```diff
@@ -71,41 +67,58 @@ const tokenize = translate => value => {
 };
 
 export class MetaTitleEditor extends Component {
-	constructor() {
+	constructor( { titleFormats } ) {
 		super();
 
-		this.state = { type: 'frontPage' };
+		this.state = {
+			titleFormats,
+			type: 'frontPage'
+		};
 
 		this.switchType = this.switchType.bind( this );
 		this.updateTitleFormat = this.updateTitleFormat.bind( this );
 	}
 
+	componentWillUpdate( nextProps ) {
+		const { siteId: prevSiteId } = this.props;
+		const { siteId: nextSiteId, titleFormats } = nextProps;
+
+		if ( nextSiteId !== prevSiteId ) {
+			// Clear all local changes when switching
+			// sites to get the new site's data
+			this.setState( { titleFormats } );
+		}
+	}
+
 	switchType( { value: type } ) {
 		this.setState( { type } );
 	}
 
 	updateTitleFormat( values ) {
-		const { setTitleFormat, siteId, translate } = this.props;
+		const { translate } = this.props;
 		const { type } = this.state;
 
 		const tokens = removeBlanks( map( values, tokenize( translate ) ) );
 
-		setTitleFormat( siteId, type, nativeToRaw( tokens ) );
+		this.setState( { titleFormats: {
+			...this.state.titleFormats,
+			[ type ]: tokens
+		} } );
 	}
 
 	render() {
 		const {
 			disabled = false,
-			translate = identity,
-			titleFormats
+			translate = identity
 		} = this.props;
 		const {
-			type
+			type,
+			titleFormats
 		} = this.state;
 
 		const validTokens = getValidTokens( translate );
 
-		const values = removeBlanks( map( rawToNative( titleFormats[ type ] ), tokenize( translate ) ) ).map(
+		const values = get( titleFormats, type, [] ).map(
```

Comment:
@gwwar I don't see this as problematic. I actually prefer the native way of chaining map calls (which could also be emulated with **lodash**'s `chain()`) but tend to use whichever feels cleanest in the situation.

that being said, this is the only place in the file where I use the native map so I went ahead and swapped it out anyway in d564a8b